library/core/src/lib.rs

```diff
@@ -151,6 +151,7 @@
 #![feature(const_slice_from_ref)]
 #![feature(const_slice_index)]
 #![feature(const_is_char_boundary)]
+#![feature(slice_split_at_unchecked)]
```

Comment:
I genuinely have no idea how this wasn't required before, since the existing implementation of `split_at` requires `split_at_unchecked` ??

Anyway, I decided to reuse this code for these methods and needed this.